src/main/java/com/github/manolo8/darkbot/core/objects/facades/SeassonPassMediator.java

```diff
@@ -0,0 +1,135 @@
+
+    @Getter
+    private CurrentLevelProgress currentLevelProgress = new CurrentLevelProgress();
+
+    @Getter
+    private CurrentLevelProgress currentSeassonLevelInfo = new CurrentLevelProgress();
+
+    @Getter
+    private boolean seassonPassAvailable = false;
+
+    private final FlashList<SeassonPassQuest> dailyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private final FlashList<SeassonPassQuest> weeklyQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private final FlashList<SeassonPassQuest> seassonQuests = FlashList.ofVector(SeassonPassQuest.class);
+    private boolean allQuestsUpdated = false;
+    private final FlashList<SeassonPassQuest> allQuests = FlashList.ofVector(SeassonPassQuest.class);
+
+    private long questDataAddr = 0;
+
+    @Override
+    public void update() {
+        if (address == 0) {
+            return;
+        }
+
+        this.allQuestsUpdated = false;
+
+        long data = API.readAtom(address + 0x60);
+        this.seassonPassAvailable = API.readBoolean(API.readAtom(data + 0x70) + 0x20);
+
+        if (!seassonPassAvailable) {
+            return;
+        }
+
+        this.currentLevelProgress.update(API.readAtom(data + 0x58));
+        this.currentSeassonLevelInfo.update(API.readAtom(data + 0x68));
+        this.questDataAddr = API.readAtom(address + 0x78);
+
+        if (questDataAddr == 0) {
+            return;
+        }
+
+        dailyQuests.update(API.readAtom(questDataAddr + 0x60));
+        weeklyQuests.update(API.readAtom(questDataAddr + 0x68));
+        seassonQuests.update(API.readAtom(questDataAddr + 0x70));
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getDailyQuests() {
+        return this.dailyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getWeeklyQuests() {
+        return this.weeklyQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getSeassonQuests() {
+        return this.seassonQuests;
+    }
+
+    public @Nullable List<? extends SeassonPassQuest> getAllQuests() {
+        if (!allQuestsUpdated && questDataAddr != 0) {
+            allQuestsUpdated = true;
+            allQuests.update(API.readAtom(questDataAddr + 0x58));
+        }
+
+        return this.allQuests;
+    }
+
+    @Getter
+    @ToString
+    public static class CurrentLevelProgress extends Auto implements SeassonPassAPI.CurrentLevelProgress {
+        private int maxProgress;
+        private int currentProgress;
+
+        @Override
+        public void update() {
+            if (address == 0) {
+                return;
+            }
+
+            this.maxProgress = API.readInt(address + 0x24);
+            this.currentProgress = API.readInt(address + 0x28);
+        }
+
+        public double getProgressPercentage() {
+            return this.currentProgress / this.maxProgress;
+        }
+    }
+
+    @Getter
+    @ToString
+    public class SeassonPassQuest extends Updatable implements SeassonPassAPI.SeassonPassQuest {

```

Comment:
```suggestion
    public static class SeassonPassQuest extends Updatable implements SeassonPassAPI.SeassonPassQuest {
```

This inner class should be static, idea would've warned about this too:
![image](https://github.com/darkbot-reloaded/DarkBot/assets/11789291/4d14dabc-3010-4b57-b6f8-679207ae0a31)

Reply:
done